### PR TITLE
Rating is loaded upon entering a place's details screen

### LINKED ISSUE #240: Landing on a place page preloads an existing rating
When a user hits a place page from the place list, if they have rated that place before, the rating they gave should be loaded. This is so when the user is taken back into the submit rating flow for a place that they have rated already they can view their previous rating rather than starting from defaults.

### PR DESCRIPTION
**Pull request by carpenterd777**
Adds a hook called `usePreviousRating` that loads the user's previous rating at the main screen component so that it can be passed to any other screen component, most notably the screen to submit a rating.

Also reorganizes the test files so that they are easier to go through.

Closes #240

### CODE REVIEW

File: src/screens/SubmitRatingScreen.tsx [hunk 1]
```diff
@@ -21,10 +23,25 @@ const SubmitRatingScreen: React.FC<SubmitRatingScreenProps> = ({
...
+					countersMap[attribute] = previousRating.navigability ?? DEFAULT_INTERIM_RATING;
+				} else if (attribute === "Sensory Aids") {
+					// TODO: change braille to sensory aids
+					countersMap[attribute] = previousRating.braille ?? DEFAULT_INTERIM_RATING;
+				} else if (attribute === "Staff Helpfulness") {
+					countersMap[attribute] =
+						previousRating.staffHelpfulness ?? DEFAULT_INTERIM_RATING;
+				} else if (attribute === "Guide Dog Friendliness") {
+					countersMap[attribute] =
+						previousRating.guideDogFriendly ?? DEFAULT_INTERIM_RATING;
```
Comment (carpenterd777) [inline src/screens/SubmitRatingScreen.tsx:42]: I'm unhappy with how this section turned out, but I'm not sure how I should approach this problem. do you all have any suggestions?

Comment (elenirotsides) [reply src/screens/SubmitRatingScreen.tsx:42]: I'm not sure what else you could do to make this smell less other than moving the logic to its own function and then calling it inside of here. But that would just move the ugly to a different place. tbh I probably would have written it the same way, curious if @aej11a or. @jlora23 have any ideas?

Comment (elenirotsides) [reply src/screens/SubmitRatingScreen.tsx:42]: I'm okay with refactoring this later, I think for now, it is totally okay @carpenterd777

Comment (carpenterd777) [reply src/screens/SubmitRatingScreen.tsx:42]: thanks. sorry to leave this in here

**Review by elenirotsides — APPROVED** [review]

**Pull request merged**